x/gov/client/rest/util.go

```diff
@@ -0,0 +1,199 @@
+	Name          string `json:"name"`
+	Password      string `json:"password"`
+	ChainID       string `json:"chain_id"`
+	AccountNumber int64  `json:"account_number"`
+	Sequence      int64  `json:"sequence"`
+	Gas           int64  `json:"gas"`
+}
+
+type postProposalReq struct {
+	Title          string    `json:"title"`           //  Title of the proposal
+	Description    string    `json:"description"`     //  Description of the proposal
+	ProposalType   string    `json:"proposal_type"`   //  Type of proposal. Initial set {PlainTextProposal, SoftwareUpgradeProposal}
+	Proposer       string    `json:"proposer"`        //  Address of the proposer
+	InitialDeposit sdk.Coins `json:"initial_deposit"` // Coins to add to the proposal's deposit
+	BaseReq        baseReq   `json:"base_req"`
+}
+
+type depositReq struct {
+	ProposalID int64     `json:"proposalID"` // ID of the proposal
+	Depositer  string    `json:"depositer"`  // Address of the depositer
+	Amount     sdk.Coins `json:"amount"`     // Coins to add to the proposal's deposit
+	BaseReq    baseReq   `json:"base_req"`
+}
+
+type voteReq struct {
+	Voter      string  `json:"voter"`      //  address of the voter
+	ProposalID int64   `json:"proposalID"` //  proposalID of the proposal
+	Option     string  `json:"option"`     //  option from OptionSet chosen by the voter
+	BaseReq    baseReq `json:"base_req"`
+}
+
+func buildReq(w http.ResponseWriter, r *http.Request, req interface{}) error {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(err.Error()))
+		return err
+	}
+	err = json.Unmarshal(body, &req)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(err.Error()))
+		return err
+	}
+	return nil
+}
+
+func (req baseReq) baseReqValidate(w http.ResponseWriter) bool {
+	if len(req.Name) == 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("Name required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+
+	if len(req.Password) == 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("Password required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+
+	// if len(req.ChainID) == 0 {
+	// 	w.WriteHeader(http.StatusUnauthorized)
+	// 	err := errors.Errorf("ChainID required but not specified")
+	// 	w.Write([]byte(err.Error()))
+	// 	return false
+	// }
+
+	if req.AccountNumber < 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("Account Number required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+
+	if req.Sequence < 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("Sequence required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+	return true
+}
+
+func (req postProposalReq) Validate(w http.ResponseWriter) bool {
+	if len(req.Title) == 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("Title required but not specified")
```

Comment:
I think the first word should be lower case in line with the go standard. 
Same goes for the rest of the similar cases beneath this. (But maybe it doesn't, since this is CLI / http errors, not errors going in the logfile)